pandas/core/generic.py

```diff
@@ -489,16 +489,51 @@ def _get_index_resolvers(self):
             d.update(self._get_axis_resolvers(axis_name))
         return d
 
-    def _get_space_character_free_column_resolvers(self):
-        """Return the space character free column resolvers of a dataframe.
+    def _get_special_character_free_column_resolvers(self, expr):
+        """Return the special character free column resolvers of a dataframe.
 
-        Column names with spaces are 'cleaned up' so that they can be referred
-        to by backtick quoting.
+        Column names with special characters are 'cleaned up' so that they can
+        be referred to by backtick quoting.
         Used in :meth:`DataFrame.eval`.
         """
-        from pandas.core.computation.common import _remove_spaces_column_name
+        from pandas.core.computation.expr import (
+            _BACKTICK_QUOTED_STRING,
+            _clean_special_characters_column_name,
+            _tokenize_string,
+        )
+
+        resolvers = [
+            (
+                _clean_special_characters_column_name(
+                    next(_tokenize_string("`%s`" % k))[1]
```

Comment:
If this is still needed should convert to str format syntax

Reply:
OK